testsuite/classlibrary/TestSimpleNumber.sc

```diff
@@ -99,7 +106,28 @@ TestSimpleNumber : UnitTest {
 		var numSecondsIn400Days = 400 * 24 * 3600;
 		var totalTime = numSecondsIn400Days + 10;
 		var actual = totalTime.asTimeString;
-		this.assert(actual == expected, "expected %, got %".format(expected, actual));
+		this.assertEquals(actual, expected, "%.asTimeString (number of days > 365 and 10 seconds)".format(totalTime));
+	}
+
+	test_asTimeString_precisionLargerThanDecimalPlaces {
+		var expected = "00:00:00.0200";
+		var totalTime = 0.015;
+		var actual = totalTime.asTimeString(precision: 0.01, decimalPlaces: 4);
+		this.assertEquals(actual, expected, "%.asTimeString".format(totalTime));
+	}
+
+	test_asTimeString_precisionSmallerThanDecimalPlaces {
+		var expected = "00:00:00.0150";
+		var totalTime = 0.015;
+		var actual = totalTime.asTimeString(precision: 0.00001, decimalPlaces: 4);
+		this.assertEquals(actual, expected, "%.asTimeString".format(totalTime));
+	}
+
+	test_asTimeString_smallNumberScientificNotation {
+		var expected = "00:00:00.0000010";
+		var totalTime = 1e-06;
+		var actual = totalTime.asTimeString(precision: 0.0000001, decimalPlaces: 7);
```

Comment:
for me anything with more zeroes than `0.001` is difficult to read  -- i would personally prefer `1e-7` here but it's not something you need to change here

Reply:
oh, sorry for the miscommunication -- i was actually talking about the float in:

```supercollider
totalTime.asTimeString(precision: 0.0000001, decimalPlaces: 7);
                                  // ^ how many zeroes?
```

i was fine with the `totalTime = 1e-6` part :)